source/nodejs/adaptivecards/src/card-elements.ts

```diff
@@ -4405,8 +4472,44 @@ export class ChoiceSetInput extends Input {
 
     protected internalRender(): HTMLElement | undefined {
         this._uniqueCategoryName = ChoiceSetInput.getUniqueCategoryName();
+        if (this.isDynamicTypeahead()) {
+            const filteredChoiceSet = new FilteredChoiceSet(
+                ChoiceSetInput._uniqueCategoryCounter,
+                this.choices,
+                this.hostConfig
+            );
+            filteredChoiceSet.render();
 
-        if (this.isMultiSelect) {
+            if (filteredChoiceSet.textInput) {
+                this._textInput = filteredChoiceSet.textInput;
+                if (this.defaultValue) {
+                    this._textInput.value = this.defaultValue;
+                }
+                if (this.placeholder && !this._textInput.value) {
+                    this._textInput.placeholder = this.placeholder;
+                    this._textInput.setAttribute("aria-label", this.placeholder);
+                }
+                this._textInput.tabIndex = this.isDesignMode() ? -1 : 0;
+                this._textInput.oninput = Utils.debounce(() => {
+                    filteredChoiceSet.processStaticChoices();
+                    this.valueChanged();
+                    if (this._textInput) {
+                        // Remove aria-label when value is not empty so narration software doesn't
+                        // read the placeholder
+                        if (this.value) {
+                            this._textInput.removeAttribute("placeholder");
+                            this._textInput.removeAttribute("aria-label");
+                        } else if (this.placeholder) {
+                            this._textInput.placeholder = this.placeholder;
+                            this._textInput.setAttribute("aria-label", this.placeholder);
+                        }
+                    }
+                }, this.hostConfig.inputs.debounceTime);
+            }
+            filteredChoiceSet.parent = this;
```

Comment:
Yes, and yes. FilteredChoiceSet is simply concerned with just the UI part.
Error label and input validation works as expected because `textInput` of `ChoiceSetInput` instance is being set with the `textInput` in the `FilteredChoiceSet` instance. This way the validation logic works in this case as well. If you look at line 4697, the `isValid` method, I'll added logic there to handle dynamic choices as well. Let me add a video in the PR description as well to show the error label.